app/controllers/katello/api/v2/content_views_controller.rb

```diff
@@ -288,7 +288,10 @@ def view_params
       if (!@content_view || !@content_view.composite?)
         attrs.push({:repository_ids => []}, :repository_ids)
       end
-      params.require(:content_view).permit(*attrs).to_h
+      result = params.require(:content_view).permit(*attrs).to_h
+      # sanitize repository_ids to be a list of integers
+      result[:repository_ids] = result[:repository_ids].map(&:to_i) if result[:repository_ids].present?
```

Comment:
```suggestion
      result[:repository_ids] = result[:repository_ids].compact.map(&:to_i) if result[:repository_ids].present?
```

Just concerned about `nil.to_i` ..

Reply:
+1 to this

Reply:
I have applied the change and re-tested.